apps/explorer/src/app/components/table/index.tsx

```diff
@@ -1,13 +1,88 @@
-import React from 'react';
+import React, { ThHTMLAttributes } from 'react';
+import classnames from 'classnames';
 
-interface TableProps {
+interface TableProps extends ThHTMLAttributes<HTMLTableElement> {
   children: React.ReactNode;
+  className?: string;
 }
 
-export const Table = ({ children }: TableProps) => {
+interface TableHeaderProps
+  extends ThHTMLAttributes<HTMLTableHeaderCellElement> {
+  children: React.ReactNode;
+  className?: string;
+}
+
+interface TableRowProps extends ThHTMLAttributes<HTMLTableRowElement> {
+  children: React.ReactNode;
+  className?: string;
+  dataTestId?: string;
+  modifier?: 'bordered' | 'background';
+}
+
+interface TableCellProps extends ThHTMLAttributes<HTMLTableCellElement> {
+  children: React.ReactNode;
+  className?: string;
+  dataTestId?: string;
+  modifier?: 'bordered' | 'background';
+}
+
+export const Table = ({ children, className, ...props }: TableProps) => {
+  const classes = classnames(className, 'overflow-x-auto whitespace-nowrap');
+  return (
+    <div className={classes}>
+      <table className="w-full" {...props}>
+        <tbody>{children}</tbody>
+      </table>
+    </div>
+  );
+};
+
+export const TableHeader = ({
+  children,
+  className,
+  ...props
+}: TableHeaderProps) => {
+  const cellClasses = classnames(className, {
+    'text-left font-normal': props?.scope === 'row',
+  });
+  return (
+    <th className={cellClasses} {...props}>
+      {children}
+    </th>
+  );
+};
+
+export const TableRow = ({
+  children,
+  className,
+  dataTestId,
```

Comment:
chore (blocking): As you  are spreading props, you don't need this. Just pass `data-testid="foo"` to the consumer and it will end up where you spread it (i.e. on the `tr`).

Few other instances of this